application/electron/src/services/files/service.file.opener.ts

```diff
@@ -198,6 +211,25 @@ class ServiceFileOpener implements IService {
         ServiceElectron.updateMenu();
     }
 
+    private _ipc_FileCheckRequest(request: IPCMessages.TMessage, response: (instance: IPCMessages.TMessage) => any) {
+        const req: IPCMessages.FileCheckRequest = request as IPCMessages.FileCheckRequest;
+        this._getFiles(req.files).then(() => {
+            response(new IPCMessages.FileCheckResponse({
+                files: this._files.slice(),
+                guid: req.guid,
+            })).catch((error: Error) => {
+                this._logger.error(`Fail to respond to files ${req.files} due error: ${error.message}`);
+            });
+            this._files = [];
```

Comment:
it doesn't make sense to use variable `_files` to deliver results of the method `_getFiles`. But in this context, it's also dangerous, because it's IPC, IPC works async always. It means could be more than 2 requests at the same time at work. It means the variable `_files` will used multiple times at the same time. 

Better just return results from `_getFiles`